packages/notebook/src/actions.tsx

```diff
@@ -1288,6 +1288,43 @@ export namespace NotebookActions {
     Private.handleState(notebook, state);
   }
 
+  /**
+   * Go to the last cell that is run or current if it is running.
+   *
+   * Note: This requires execution timing to be toggled on or this will have
+   * no effect.
+   *
+   * @param notebook - The target notebook widget.
+   */
+  export function selectLastRunCell(notebook: Notebook): void {
+    let latestTime: Date = null;
+    let latestCellIdx: number = null;
+    notebook.widgets.forEach((cell, cellIndx) => {
+      if (cell.model.type === 'code') {
+        const execution = (cell as CodeCell).model.metadata.get('execution');
+        if (
+          execution &&
+          typeof execution === 'object' &&
+          'iopub.status.busy' in execution
```

Comment:
Nitpick: Could these two lines be replaced by `execution['iopub.status.busy'] !== undefined` ?

Reply:
This was done. I will note that I had to cast it to a JSONObject before which is slightly less safe. If execution was set to `3` this will now throw.

Reply:
Good point. I think that using `@phosphor/coreutils/JSONExt.isObject()` will likely be the most robust method here (both in JS and for typings).

For reference, `typeof null === 'object'` 😕

Reply:
(Note: I'm not implying that the previous code was wrong, just that `typeof X === 'object'` is so counter-intuitive most of the time that I've made it a rule of thumb to simply avoid it)

Reply:
Fixed this anyway. Sadly, isObject fails to return a falsy value with `JSONExt.isObject(undefined)`